packages/client/lib/OpenID4VCIClient.ts

```diff
@@ -71,37 +71,55 @@ export class OpenID4VCIClient {
     return this._serverMetadata;
   }
 
-  public createAuthorizationRequestUrl({ clientId, codeChallengeMethod, codeChallenge, redirectUri, scope }: AuthorizationRequestOpts): string {
-    if (!scope) {
-      throw Error('Please provide a scope. authorization_details based requests are not supported at this time');
+  public createAuthorizationRequestUrl({ clientId, codeChallengeMethod, codeChallenge, authorizationDetails, redirectUri, scope }: AuthorizationRequestOpts): string {
+
+    // Scope and authorization_details can be used in the same authorization request
+    // https://datatracker.ietf.org/doc/html/draft-ietf-oauth-rar-23#name-relationship-to-scope-param
+    if (!scope && !authorizationDetails) {
+      throw Error('Please provide a scope or authorization_details');
     }
 
     if (!this._serverMetadata?.openid4vci_metadata?.authorization_endpoint) {
       throw Error('Server metadata does not contain authorization endpoint');
     }
 
     // add 'openid' scope if not present
-    if (!scope.includes('openid')) {
+    if (scope && !scope.includes('openid')) {
       scope = `openid ${scope}`;
     }
 
+    this.addLocations(authorizationDetails );
+
     const queryObj: AuthorizationRequest = {
       response_type: ResponseType.AUTH_CODE,
       client_id: clientId,
       code_challenge_method: codeChallengeMethod,
       code_challenge: codeChallenge,
+      authorization_details: JSON.stringify(authorizationDetails),
       redirect_uri: redirectUri,
       scope: scope,
     };
 
     const authRequestUrl = convertJsonToURI(queryObj, {
       baseUrl: this._serverMetadata.openid4vci_metadata.authorization_endpoint,
-      uriTypeProperties: ['redirect_uri', 'scope'],
+      uriTypeProperties: ['redirect_uri', 'scope', 'authorization_details']
     });
 
     return authRequestUrl;
   }
 
+  private addLocations(authorizationDetails?: AuthorizationDetails | AuthorizationDetails[]): void {
+    if (authorizationDetails && this.serverMetadata.openid4vci_metadata?.authorization_server) {
```

Comment:
Have you looked at the other property pointing to an AS which we are seeing in the wild?

Reply:
According to the comment, authorization_server is the property used in the wild and the specs are very specific to use it here.

Reply:
Okay, so why are you then only using the one which is not spec compliant here?

```` typescript
authorization_server?: string; //NON-SPEC compliant, but used by some issuers. URL of the AS. This URL MUST use the https scheme and MAY contain port, path and query parameter components.
  // TODO: The above authorization_server being used in the wild, serves roughly the same purpose as the below spec compliant endpoint. Look at how to use authorization_server as authorization_endpoint in case it is present
authorization_endpoint?: string;````

Reply:
According to the [spec](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#section-5.1.1-5): "If the Credential Issuer metadata contains an authorization_server parameter, the authorization detail's locations common data field MUST be set to the Credential Issuer Identifier value".

Reply:
Okay than it probably changed in later specs. Still we have both values in the wild. Meaning we need to accomodate for both

Reply:
Fixed!